src/vpart_range.h

```diff
@@ -35,7 +36,8 @@ class vehicle_part_iterator
             return range_.get();
         }
         void skip_to_next_valid( size_t i ) {
-            while( i < range().part_count() && !range().contained( i ) ) {
+            while( i < range().part_count() &&
+                   !range().matches( i, static_cast<part_status_flag>( 0 ) ) ) {
```

Comment:
Why are you using a fixed value 0 here? If the parameter is always the same, why have it at all? The ranges should store their own `part_status_flag` value (and already do so when needed).

Reply:
Not sure what I was thinking here.
Revised.